filterExprParser.go

```diff
@@ -0,0 +1,1160 @@
+// And due to the complicated FEOperand op, we have to do char by char match so we can catch the not-matched case
+// and go to the other type of operands
+
+type FEOpChar struct {
+	Equal       *bool `( @"=" |`
+	LessThan    *bool `@"<" |`
+	GreaterThan *bool `@">" )`
+}
+
+type FECompareOp struct {
+	OpChars0 *FEOpChar `@@`
+	OpChars1 *FEOpChar `[ @@ ]`
+}
+
+func (feo *FECompareOp) IsEqual() bool {
+	// =
+	return feo.OpChars0 != nil && feo.OpChars0.Equal != nil && feo.OpChars0 != nil && feo.OpChars1 == nil
+}
+
+func (feo *FECompareOp) IsNotEqual() bool {
+	// <>
+	return feo.OpChars0 != nil && feo.OpChars0.LessThan != nil && feo.OpChars1 != nil && feo.OpChars1.GreaterThan != nil
+}
+
+func (feo *FECompareOp) IsGreaterThan() bool {
+	// >
+	return feo.OpChars0 != nil && feo.OpChars0.GreaterThan != nil && feo.OpChars1 == nil
+}
+
+func (feo *FECompareOp) IsGreaterThanOrEqualTo() bool {
+	// >=
+	return feo.OpChars0 != nil && feo.OpChars0.GreaterThan != nil && feo.OpChars1 != nil && feo.OpChars1.Equal != nil
+}
+
+func (feo *FECompareOp) IsLessThan() bool {
+	// <
+	return feo.OpChars0 != nil && feo.OpChars0.LessThan != nil && feo.OpChars1 == nil
+}
+
+func (feo *FECompareOp) IsLessThanOrEqualTo() bool {
+	// <=
+	return feo.OpChars0 != nil && feo.OpChars0.LessThan != nil && feo.OpChars1 != nil && feo.OpChars1.Equal != nil
+}
+
+func (feo *FECompareOp) String() string {
+	if feo.IsEqual() {
+		return "="
+	} else if feo.IsNotEqual() {
+		return "!="
+	} else if feo.IsGreaterThan() {
+		return ">"
+	} else if feo.IsGreaterThanOrEqualTo() {
+		return ">="
+	} else if feo.IsLessThan() {
+		return "<"
+	} else if feo.IsLessThanOrEqualTo() {
+		return "<="
+	}
+	return "?? (FECompareOp)"
+}
+
+func (f *FECompareOp) OutputExpression(lhs Expression, rhs Expression) (Expression, error) {
+	if f.IsEqual() {
+		return EqualsExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	} else if f.IsNotEqual() {
+		return NotEqualsExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	} else if f.IsGreaterThan() {
+		return GreaterThanExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	} else if f.IsGreaterThanOrEqualTo() {
+		return GreaterEqualsExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	} else if f.IsLessThan() {
+		return LessThanExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	} else if f.IsLessThanOrEqualTo() {
+		return LessEqualsExpr{
+			Lhs: lhs,
+			Rhs: rhs,
+		}, nil
+	}
+	return nil, fmt.Errorf("Invalid FECompareOp %v", f.String())
+}
+
+type FECheckOp struct {
+	Exists  *bool `@"EXISTS" | ( "IS"`
+	Not     *bool `[ @"NOT" ]`
+	Null    *bool `( @"NULL" |`
+	Missing *bool `@"MISSING" ) )`
+}
+
+func (feco *FECheckOp) IsExists() bool {
+	return feco.Exists != nil && *feco.Exists == true
+}
+
+func (feco *FECheckOp) isNot() bool {
+	return feco.Not != nil && *feco.Not == true
+}
+
+func (feco *FECheckOp) IsMissing() bool {
+	return !feco.isNot() && feco.Missing != nil && *feco.Missing == true
+}
+
+func (feco *FECheckOp) IsNotMissing() bool {
+	return !feco.IsMissing()
```

Comment:
not correct. should be isNot() && missing = true

Reply:
Done